lib/racer-client.coffee

```diff
@@ -88,10 +94,13 @@ class RacerClient
 
   parse_single: (line) ->
     matches = []
-    rcrgex = /MATCH (\w*)\,\d*\,\d*\,([^\,]*)\,(\w*)\,.*\n/mg
+    rcrgex = /MATCH (\w*)\,(\d*)\,(\d*)\,([^\,]*)\,(\w*)\,.*\n/mg
     while match = rcrgex.exec(line)
-      if match?.length > 2
-        candidate = {word: match[1], file: "this", type: match[3]}
-        candidate.file = path.basename(match[2]) if path.extname(match[2]) != ".racertmp"
+      if match?.length > 4
+        candidate = {word: match[1], line: parseInt(match[2], 10), column: parseInt(match[3], 10), filePath: match[4], file: "this", type: match[5]}
```

Comment:
Maybe it would be better to prefix those special fields additions, so that they never collide with the Autocomplete+ suggestion format (and that we can easily distinguish them and note they are for our own use)?